backend/googlecloudstorage/googlecloudstorage.go

```diff
@@ -681,9 +697,15 @@ func (f *Fs) list(ctx context.Context, bucket, directory, prefix string, addBuck
 			}
 			// is this a directory marker?
 			if isDirectory {
-				continue // skip directory marker
+				// Don't insert the root directory
+				if remote == directory {
+					continue
+				}
+				// process directory markers as directories
+				remote = strings.TrimRight(remote, "/")
```

Comment:
@ncw  This is causing `rclone lsd -R` to throw out the following errors when encountering directory markers:
```
2023/05/26 22:02:24 ERROR : foo: Entry doesn't belong in directory "foo" (too short) - ignoring
           0 2023-05-26 22:02:24        -1 foo/bar
           0 2023-05-26 22:02:24        -1 foo/baz
2023/05/26 22:02:24 ERROR : httpdocs: Entry doesn't belong in directory "httpdocs" (too short) - ignoring
           0 2023-05-26 22:02:24        -1 httpdocs/admin
           0 2023-05-26 22:02:24        -1 httpdocs/bar
           0 2023-05-26 22:02:24        -1 httpdocs/cache
```

In the above example, `foo/` and `httpdocs/` are both GCS directory markers. They fall into https://github.com/rclone/rclone/blob/e43b5ce/fs/list/list.go#L79 : `prefix` is set to `foo/` when the `remote` is `foo`. 

Also, if the root directory is a directory marker, it falls into https://github.com/rclone/rclone/blob/e43b5ce/fs/list/list.go#L82 ,  triggering an error like this:
```
2023/05/26 22:08:30 ERROR : : Entry doesn't belong in directory "" (same as directory) - ignoring
```

I think we need to handle directory markers properly in `filterAndSortDir()`. 

A possible solution is to prepend `/` to every entry, including directory markers, e.g.
```
/httpdocs (directory-marker)
/httpdocs/admin (non directory-marker subdir)
/httpdocs/cache (non directory-marker subdir)
```
When running `rclone lsd -r httpdocs`, the output would be:
```
/ (directory-marker)
/admin (non directory-marker subdir)
/cache (non directory-marker subdir)
```

Alternatively, we could show the entries as follows:
```
httpdocs/
httpdocs/admin/
httpdocs/cache/
```
and
```
/
admin/
cache/
```

Thoughts or suggestions? 🙏 

Note: I added this comment to the original commit but I realised that it's probably not visible. So moving it here. Hope it's okay.